File: k-kbk/6주차/9996.py
import sys

input = sys.stdin.readline

n = int(input())

pattern = input()
start, end = pattern.split("*")

for _ in range(n):
    name = input()
    print(
        "DA"
        if len(name) >= len(pattern) - 1
        and name.startswith(start)
        and name.endswith(end)
        else "NE"
    )

File: keemsebin/6주차/9996.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];

rl.on("line", function (line) {
  input.push(line);
}).on("close", function () {
  let N = Number(input[0]);
  let [start, end] = input[1].split("*");
  let result = [];
  let files = input.slice(2);

  for (let i = 0; i < N; i++) {
    let file = files[i].trim();

    if (file.length < start.length + end.length) {
      result.push("NE");
      continue;
    }
    if (file.startsWith(start) && file.endsWith(end)) {
      let middleLength = file.length - start.length - end.length;
      if (middleLength >= 0) {
        result.push("DA");
      } else {
        result.push("NE");
      }
    } else {
      result.push("NE");
    }
  }
  console.log(result.join("\n"));
});
